### DISCUSSION TITLE
A question about how to do the link prediction task

### DISCUSSION THREAD

**Discussion by HanChen-HUST** [pyg-team/pytorch_geometric#5808] [Q&A]
```python
from sklearn.metrics import roc_auc_score
from torch_geometric import transforms as T
split = T.RandomLinkSplit(num_val=0., num_test=0., is_undirected=True, add_negative_train_samples=False)
#negative sampling by torch_geometric.transforms
data_z = split(z)[0]
src = z[data_z.edge_label_index[0]]
dst = z[data_z.edge_label_index[1]]
out = (src * dst).sum(dim=-1).view(-1).sigmoid()
result = roc_auc_score(data_z.edge_label.cpu().numpy(), out.cpu().numpy())
print(result)
```
Hi@rusty1s ,I am using a unsupervised learning algorithm to get the node embedding to do the link prediction downstream task.z is the node embedding which I get,I want to test the whole dataset to get the AUC score,so the negative sampling I set num_val=0. and 
 num_test=0. By the code above,do you think is a proper way to do this,I am quited confused,much apperciate!

**Comment by rusty1s** [comment]
Yes, if you don't want to do any evaluation on unseen edges, you can set `num_val=0` and `num_test=0`.

**Reply by HanChen-HUST** [reply]
I consider num_val and num_test are the numbers of the validation and test dataset,is it has any realation with the unseen or can be called faked edge？

**Reply by rusty1s** [reply]
`num_val` and `num_test` refer to the number of original edges you will use as the positive links for evaluation.
